test/index.test.js

```diff
@@ -31,46 +31,90 @@ test('Should not expose jaeger api when explicitly set to false', async ({ teard
   is(fastify.hasRequestDecorator('jaeger'), false)
 })
 
-test('Should initialize plugin with default configuration', async ({ teardown, is, has }) => {
-  delete require.cache[require.resolve('jaeger-client')]
-  delete require.cache[require.resolve(require('path').join(__dirname, '../index'))]
-  require('jaeger-client')
-  const initTracerSpy = spy(require.cache[require.resolve('jaeger-client')].exports.initTracer)
-  require.cache[require.resolve('jaeger-client')].exports.initTracer = initTracerSpy
-
-  const fastify = createFastify()
-
-  teardown(async () => {
-    delete require.cache[require.resolve('jaeger-client')]
-    delete require.cache[require.resolve(require('path').join(__dirname, '../index'))]
-    await fastify.close()
-  })
-
-  fastify.get('/', (req, reply) => {
-    reply.code(200).send({ hello: 'world' })
+test('Should initialize plugin with correct configuration', async (child) => {
+  const serviceName = 'test'
+
+  const setupChildTest = (pluginOpts) => {
+    const clear = () => {
+      delete require.cache[require.resolve('jaeger-client')]
+      delete require.cache[require.resolve(require('path').join(__dirname, '../index'))]
+    }
+
+    clear()
+    require('jaeger-client')
+    const initTracerSpy = spy(require.cache[require.resolve('jaeger-client')].exports.initTracer)
+    require.cache[require.resolve('jaeger-client')].exports.initTracer = initTracerSpy
+
+    const fastify = createFastify()
+    fastify.get('/', (req, reply) => {
+      reply.code(200).send({ hello: 'world' })
+    })
+
+    fastify.register(require('../index'), pluginOpts)
+
+    return {
+      fastify,
+      initTracerSpy,
+      teardown: async () => {
+        clear()
+        await fastify.close()
+      }
+    }
+  }
+
+  child.test('should use default config when no configuration is provided', async (ct) => {
```

Comment:
Can we have these set of tests as independent tests or root tests on its own?

Reply:
✅